Fleury/Fleury.java

```diff
@@ -0,0 +1,66 @@
+// This code is contributed by Riyazul555
+
+// Time Complexity  =  O(V + E)  where V is the number of vertices and E is the number of edges.
+// Space Complexity =  O(V + E)  where V is the number of vertices and E is the number of edges.
+
+import java.io.*;
+import java.util.*;
+
+public class EulerPath {
+
+    static final int nMax = 100005;
+
+    static void euler(int nod, ArrayList<Pair<Integer, Integer>>[] G, ArrayList<Integer> e, BitSet viz) {
+        while (!G[nod].isEmpty()) {
+            Pair<Integer, Integer> it = G[nod].remove(G[nod].size() - 1);
+            if (!viz.get(it.second)) {
+                viz.set(it.second);
+                euler(it.first, G, e, viz);
+            }
+        }
+        e.add(nod);
+    }
+
+    public static void main(String[] args) throws IOException {
+        BufferedReader fin = new BufferedReader(new FileReader(args[0]));
+        BufferedWriter fout = new BufferedWriter(new FileWriter(args[1]));
+
+        StringTokenizer st = new StringTokenizer(fin.readLine());
+        int n = Integer.parseInt(st.nextToken());
+        int m = Integer.parseInt(st.nextToken());
+
+        ArrayList<Pair<Integer, Integer>>[] G = new ArrayList[nMax];
+        for (int i = 0; i < nMax; i++) {
+            G[i] = new ArrayList<>();
+        }
+
+        for (int i = 1; i <= m; i++) {
+            st = new StringTokenizer(fin.readLine());
+            int x = Integer.parseInt(st.nextToken());
+            int y = Integer.parseInt(st.nextToken());
+            G[x].add(new Pair<>(y, i));
+            G[y].add(new Pair<>(x, i));
+        }
+
+        ArrayList<Integer> e = new ArrayList<>();
+        BitSet viz = new BitSet(5 * nMax);
```

Comment:
`viz` is a frequency array for the edges of the graph, so let's initialize its size with `m`.